lua/vex_library/modules/server/e2helperfuncs.lua

```diff
@@ -84,8 +85,15 @@ vex.luaTableToE2 = function(tbl)
             t,t_types = s,stypes
         else continue end -- Don't do non-string/number keys
         size = size + 1
-        if V == tbl then
-            t[K],t_types[K] = output,"t"
+        if istable(V) then
+            t_types[K] = "t"
+            if V == tbl then
+                t[K] = output
+            elseif arrayOptimization and table.IsSequential(V) then
+                t[K],t_types[K] = V,"r"
+            else
+                t[K] = vex.luaTableToE2(V)
```

Comment:
Not passing it into recursion?

Reply:
wdym

Reply:
The flag is lost after the first depth. For example if you have a table and another table inside that is supposed to have arrays within it, then it would treat it as a table and not an array. I will fix this.

Reply:
Agh I'm blind, thanks